modules/local/dada2_quality.nf

```diff
@@ -13,12 +13,61 @@ process DADA2_QUALITY {
     output:
     path "${meta}_qual_stats.pdf"            , emit: pdf
     tuple val(meta), path("*_qual_stats.tsv"), emit: tsv
+    path "versions.yml"                      , emit: versions
     path "*.args.txt"                        , emit: args
+    path "*plotQualityProfile.txt"           , emit: warning
 
     script:
     def args = task.ext.args ?: ''
     """
-    dada_quality.r "${meta}_qual_stats" $args
-    echo 'plotQualityProfile\t$args' > "plotQualityProfile.args.txt"
+    #!/usr/bin/env Rscript
+
+    suppressPackageStartupMessages(library(dada2))
+    suppressPackageStartupMessages(library(ShortRead))
+
+    readfiles <- sort(list.files(".", pattern = ".fastq.gz", full.names = TRUE))
+
+    #make list of number of sequences
+    readfiles_length <- countLines(readfiles) / 4
+
+    #use only the first x files when read number gets above 2147483647, read numbers above that do not fit into an INT and crash the process!
+    if ( sum(readfiles_length) > 2147483647 ) {
+        for (i in length(readfiles):1) {
+            if ( sum(readfiles_length[1:i]) <= 2147483647 ) {
+                max_files = i
+                break
+            }
+        }
+        write.table(max_files, file = paste0("WARNING Only ",max_files," of ",length(readfiles)," files and ",sum(readfiles_length[1:max_files])," of ",sum(readfiles_length)," reads were used for ${meta} plotQualityProfile.txt"), row.names = FALSE, col.names = FALSE, quote = FALSE, na = '')
+        readfiles <- readfiles[1:max_files]
+    } else {
+        max_files <- length(readfiles)
+        write.table(max_files, file = paste0(max_files," files were used for ${meta} plotQualityProfile.txt"), row.names = FALSE, col.names = FALSE, quote = FALSE, na = '')
+    }
+
+    plot <- plotQualityProfile(readfiles$args)
+    data <- plot\$data
+
+    df <- data.frame(Cycle=character(), Count=character(), Median=character(), stringsAsFactors=FALSE)
+    cycles <- sort(unique(data\$Cycle))
```

Comment:
Is the "\" supposed to be here?

Reply:
sry, cannot see a `""`, waht do you mean?

Reply:
Yea sorry, I meant the slash
I was wondering if `data\$Cycle` needs to be used for using R here or whether it can be `data$Cycle`

Reply:
That slash is required because its in the nextflow script. If that would be outside of the nextflow script block, than it may not have a slash at that position, that is true.